package.json

```diff
@@ -17,6 +17,7 @@
     "react-redux": "^7.0.3",
     "react-scripts": "^3.0.1",
     "redux": "^4.0.1",
+    "redux-batched-actions": "^0.4.1",
```

Comment:
This is not needed. Tried it, doesn't seem to work.

Reply:
I can't really tell which one is better. @vincerubinetti, can you write down the pros and cons of each approach?

Reply:
#### The inherent problem
The url must stay in sync with the relevant bits of global state. Any time the user changes the global state (searches for a new source node, checks a new metapath, swaps source/target nodes, etc), the url should update. However, when the user navigates back/forward in the history, the state must be updated from the url. In a typical React setup, components listening to changes in the global state have no way to tell whether the state change came from a user interaction or from updating the state from the url. This leads to pushing multiple redundant history entries, pushing non-sensical interim states to the history (like the source/target nodes state has updated but the metapaths hasn't yet), and may other issues.

#### Approach 1
Uses a single redux action `setState` to allow setting multiple global vars "at the same time" (without triggering multiple state and component updates). Not redux-idiomatic.

#### Approach 2
Moves the url update function to the redux reducers file, and passes a flag through actions saying whether to update the url.

Reply:
@vincerubinetti: Thanks for the explanation. I read the code again and lean toward approach #2 too. What do you think @arielsvn?

Reply:
Same, I think the other approach is better

Reply:
I agree. This approach is terrible :octopus:

Reply:
Still it was interesting to be able to compare them side by side, thanks for mentioning me. Enjoyed taking a peek :)

Reply:
Thanks to both of you! @vincerubinetti: I will take a closer look at the other PR. Please feel free to close this PR.